### PR TITLE
refactor execution context

### PR DESCRIPTION
**Pull request by iurimatias**
stops relying on global execution context and instead pass it as parameter

### CODE REVIEW

File: src/app/files/compiler-metadata.js [hunk 1]
```diff
@@ -97,7 +97,7 @@ class CompilerMetadata extends Plugin {
       var provider = self.fileManager.currentFileProvider()
       var path = self.fileManager.currentPath()
       if (provider && path) {
-        executionContext.detectNetwork((err, { id, name } = {}) => {
+        self.executionContext.detectNetwork((err, { id, name } = {}) => {
```
Comment (GrandSchtroumpf) [inline src/app/files/compiler-metadata.js]: you should be able to use `this` here I think

File: src/app/tabs/runTab/model/dropdownlogic.js [hunk 2]
```diff
@@ -106,25 +107,25 @@ class DropdownLogic {
 
   fromWei (value, doTypeConversion, unit) {
     if (doTypeConversion) {
-      return executionContext.web3().fromWei(typeConversion.toInt(value), unit || 'ether')
+      return Web3.utils.fromWei(typeConversion.toInt(value), unit || 'ether')
```
Comment (GrandSchtroumpf) [inline src/app/tabs/runTab/model/dropdownlogic.js]: Why not using ethers.js instead as we're moving out of Web3 to ethers.js ?

File: src/app/ui/txLogger.js [hunk 3]
```diff
@@ -205,7 +204,7 @@ function log (self, tx, receipt) {
   }
 }
 
-function renderKnownTransaction (self, data) {
+function renderKnownTransaction (self, data, executionContext) {
```
Comment (GrandSchtroumpf) [inline src/app/ui/txLogger.js]: Why not using `this` or `self` for `executionContext` as you're doing in the other files ?

File: src/app/ui/universal-dapp-ui.js [hunk 4]
```diff
@@ -180,13 +180,13 @@ UniversalDAppUI.prototype.getCallButton = function (args) {
           cb(txFeeText, priceStatus)
         },
         (cb) => {
-          executionContext.web3().eth.getGasPrice((error, gasPrice) => {
+          self.executionContext.web3().eth.getGasPrice((error, gasPrice) => {
```
Comment (GrandSchtroumpf) [inline src/app/ui/universal-dapp-ui.js]: Can you not use `this` here ?

File: src/app/ui/universal-dapp-ui.js [hunk 5]
```diff
@@ -264,7 +264,7 @@ UniversalDAppUI.prototype.getCallButton = function (args) {
               }
             }
             if (lookupOnly) {
-              var decoded = decodeResponseToTreeView(executionContext.isVM() ? txResult.result.execResult.returnValue : ethJSUtil.toBuffer(txResult.result), args.funABI)
+              var decoded = decodeResponseToTreeView(self.executionContext.isVM() ? txResult.result.execResult.returnValue : ethJSUtil.toBuffer(txResult.result), args.funABI)
```
Comment (GrandSchtroumpf) [inline src/app/ui/universal-dapp-ui.js]: Can you not use `this` here instead of `self` ?

File: src/lib/cmdInterpreterAPI.js [hunk 6]
```diff
@@ -62,14 +62,14 @@ class CmdInterpreterAPI {
   debug (hash, cb) {
     var self = this
     delete self.d
-    executionContext.web3().eth.getTransaction(hash, (error, tx) => {
+    self.executionContext.web3().eth.getTransaction(hash, (error, tx) => {
```
Comment (GrandSchtroumpf) [inline src/lib/cmdInterpreterAPI.js]: Use `this` instead

File: src/lib/cmdInterpreterAPI.js [hunk 7]
```diff
@@ -62,14 +62,14 @@ class CmdInterpreterAPI {
...
       if (error) return cb(error)
       var debugSession = new RemixDebug({
         compilationResult: () => {
           return self._deps.compilersArtefacts['__last'].getData()
         }
       })
-      debugSession.addProvider('web3', executionContext.web3())
+      debugSession.addProvider('web3', self.executionContext.web3())
```
Comment (GrandSchtroumpf) [inline src/lib/cmdInterpreterAPI.js]: `this`

File: src/app/tabs/runTab/model/dropdownlogic.js (hunk 2 above)
Comment (iurimatias) [reply src/app/tabs/runTab/model/dropdownlogic.js]: I just wanted to get rid of the using executionContext here avoiding breaking changes as maximum as possible. I agree later we can refactor this to either use web3-utils directly or something else like etherjs, it's quite possible this whole method might disapear though so optimizing now could be premature

File: src/app/ui/txLogger.js (hunk 3 above)
Comment (iurimatias) [reply src/app/ui/txLogger.js]: keeping the same behaviour for now

**Comment by iurimatias** [comment]
@LianaHus it's fixed now

**Comment by iurimatias** [comment]
closing since the changes in this PR were included in #2506

**Pull request closed without merging**